src/java/org/apache/cassandra/concurrent/SEPExecutor.java

```diff
@@ -260,23 +282,64 @@ public long getCompletedTaskCount()
 
     public int getActiveTaskCount()
     {
-        return maxWorkers - workPermits(permits.get());
+        return maxWorkers.get() - workPermits(permits.get());
+    }
+
+    public int getCorePoolSize()
+    {
+        return 0;
+    }
+
+    public void setCorePoolSize(int newCorePoolSize)
+    {
+        throw new IllegalArgumentException("Cannot resize core pool size of SEPExecutor");
     }
 
     @Override
     public int getMaximumPoolSize()
     {
-        return maxWorkers;
+        return maxWorkers.get();
+    }
+
+    @Override
+    public void setMaximumPoolSize(int newMaximumPoolSize)
```

Comment:
synchronised?

Reply:
can make it synchronized - I was relying on maxWorkers CAS to protect against concurrent updates, but it's unlikely to be called frequently and makes it more obvious.

Reply:
We're throwing `ISE` on concurrent updates to `maxWorkers` though?  Admittedly this would be a hard race to encounter given current usages.  Don't mind what we choose, just asking the questions.